File: TPSI/Compiti/Rizzolo_lorenzo/rizzolo_bit_di_parita.c
#include <stdio.h>
#define bit 5

void stamp(int n[bit]){
    for(int i=0 ; i<bit ; i++){
        printf("%d", n[i]);
    }
}

main(){
    int num[bit], somma=0,bit_di_parita=0;
    for(int i=0 ; i<bit ; i++){
        printf("inserisci numero-> "); scanf("%d",&num[i]);
        somma=somma+num[i];
    }
    stamp(num); printf("\n");
    if(somma%2==0){
        bit_di_parita=0;
    }else{
        bit_di_parita=1;
    }
    printf("bit di parita'= %d", bit_di_parita);
    printf("\n");
    stamp(num);printf("%d", bit_di_parita);

}

File: TPSI/Compiti/Rizzolo_lorenzo/rizzolo_bit_di_parita.py
somma=0
somma=int(somma)
v=[]
bi=input("quanti bit vuoi inserire? -> ")
bi=int(bi)
for i in range(bi):
    n=input("inserisci bit-> ")
    n=int(n)
    somma=somma+n
    v.append(n)
if somma%2==0:
    b='0' 
    print("bit di parita'= "+b)
    
else:
    print("bit di parita'= "+1)
    b=1
    b=int(b)

for i in range(bi):
    print(v[i], end="")
print(b)